src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -76,11 +76,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+
+    /*String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
     Message message = new Message(user, text);
     datastore.storeMessage(message);
 
+    response.sendRedirect("/user-page.html?user=" + user);*/
+
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    String textWithImagesReplaced = userText.replaceAll(regex, replacement);
+
+    Message message = new Message(user, textWithImagesReplaced);
+    datastore.storeMessage(message);
+
     response.sendRedirect("/user-page.html?user=" + user);
+
```

Comment:
I'm not sure if you're done with your commits, but remove these extra spaces when you get the chance!